WickedEngine/wiGraphics.h

```diff
@@ -479,22 +495,9 @@ namespace wiGraphics
 		} type = MEMORY_BARRIER;
 		union
 		{
-			struct Memory
-			{
-				const GPUResource* resource;
-			} memory;
-			struct Image
-			{
-				const Texture* texture;
-				IMAGE_LAYOUT layout_before;
-				IMAGE_LAYOUT layout_after;
-			} image;
-			struct Buffer
-			{
-				const GPUBuffer* buffer;
-				BUFFER_STATE state_before;
-				BUFFER_STATE state_after;
-			} buffer;
+			struct Memory memory;
+			struct Image image;
+			struct Buffer buffer;
```

Comment:
This is interesting, what errors were You getting on Linux trying to compile the old version?

Reply:
Ok, tried it in Godbolt, looks like it complaining about declaring types in anonymous union.
If the Memory/Image/Buffer types aren't used anywhere else, that union could be rewritten as:
```cpp
		union
		{
          	struct 
            {
                const GPUResource* resource;
            } memory;
            struct
            {
                const Texture * texture;
                IMAGE_LAYOUT layout_before;
                IMAGE_LAYOUT layout_after;
            } image;
            struct 
            {
                const GPUBuffer *buffer;
                BUFFER_STATE state_before;
                BUFFER_STATE state_after;
            } buffer;
		};
```

Reply:
Probably. Most of what I did was to get it to compile, so that we can start working on add the Linux layer to files like wiRawInput.cpp. There is definitely some syntax polish that can be done.